Michaelberk/model registry 2 (#10537)
mlflow/mlflow: docs/source/getting-started/model-registry/index.rst

Tutorial Overview
=================

The MLflow Model Registry has several core components:

* A **Centralized Model Store** is a single location for your MLflow models, facilitating model versioning, sharing, and deployment in a consistent and efficient manner.
* A **Set of APIs** that allow you to programmatically create, read, update, and delete models.
* A **GUI** that allows you to manually view and manage models in the centralized model store.

The MLflow Model Resigry provides some additional functionality that is relevant to model 
development and deployment:

* **Lineage** encompasses the origin and development history of a model, including information like source code, data, and parameter configuration.
* **Model Versioning** refers to logging different iterations of a model to facilitate comparison and serving. By default, models are versioned with a monotonically increasing ID, but you can also alias model versions.

> * **Model Aliasing** involves assigning readable names to a model, simplifying model identification and management.

Reviewer: ```suggestion
* **Model Aliasing** allows you to assign mutable, named references to particular versions of a model, simplifying model deployment.
```

> * **Model Tagging** allows users to attach custom key-value pairs to a model, facilitating categorization and retrieval.

Reviewer: ```suggestion
* **Model Tagging** allows users to label models with custom key-value pairs, facilitating documentation and categorization.
```

* **Model Annotations** are descriptive notes added to a model. 

In this tutorial, you will get up and running with the MLflow model registry in the least amount of
steps possible. The topics in this tutorial cover:

* Registering a model programmatically to the Model Registry while logging.

> * Viewing the model in the MLflow UI.

Reviewer: ```suggestion
* Viewing the registered model in the MLflow UI.
```

> * Referencing a logged model.

Reviewer: Do you actually talk about this?

> * Loading a logged model for inference.

Reviewer: ```suggestion
* Loading a registered model for inference.
```

> * Aliasing model versions.

Reviewer: I don't think you actually introduce the concept of model versions yet in this tutorial. Wondering if there are other ways to talk about this instead.

Author: Added a definition in step 3
